lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -355,6 +400,16 @@ def cleanup():
                 logger.error("Service previously started but now misses the snap.")
                 return
 
+        # Store the current perf. profile we are applying
+        self.opensearch_config.apply_performance_profile(self.opensearch.perf_profile)
+        self._apply_profile_templates_event.emit()
+        if self.unit.is_leader():
+            self.peers_data.put(
```

Comment:
I prefer we enrich the `deployment-description` object and follow its flow by adding a field `profile: str` in the said object.

Reply:
@Mehdi-Bendriss I think it is not clear what this entry means vs. the `deployment-description`.

DeploymentDescription carries the directive: everyone must be on profile="YY".

This setting is used to tell us which profile this unit we have right now. If:

```
self.peers_data.get(Scope.UNIT, PERFORMANCE_PROFILE) != DeploymentDescription.get(PERFORMANCE_PROFILE)
```

Then we need to change the settings and trigger a restart.

Reason being: it is far easier to keep notes of storing our current profile state on the peer databag than figuring it out from the current deployment.

Reply:
You are not storing the flag in the unit peer data, you are storing it in the app data bag: `Scope.APP,`

Reply:
Yes, that has been fixed within `opensearch_performance_profile` module.